### DISCUSSION TITLE
请问一下可以设置打开编辑器，默认选中左侧边栏指定的某一项吗

### DISCUSSION THREAD

**Discussion by bytemain** [opensumi/core#1692] [Q&A] [answered]
![CleanShot 2022-09-21 at 14 40 21](https://user-images.githubusercontent.com/13938334/191432422-c646028c-4217-44d3-b340-46f579f2fb5c.png)

 请问一下左侧边栏可以修改成默认选中指定的某个模块吗

**Comment by bytemain** [comment, marked as answer]
这一块在 layout 配置里应该是属于 left 插槽，IMainLayoutService.getTabbarService('left') 就可以拿到这个 tabbar 的控制权了。
可以设置选中哪一个 item。

```ts
import { IMainLayoutService } from '@opensumi/ide-main-layout';
@Injectable()
class XXX {
  @Autowired(IMainLayoutService)
  mainLayoutService: IMainLayoutService;

  func() {
     const tabbarService = this.mainLayoutService.getTabbarService('left');
     tabbarService.currentContainerId = 'explorer';
  }
}

```

这里的 explorer 可以换成你想要的。

插件注册视图的时候都会有一个 id，可以去看这个插件的 package.json。里面会声明它注册的按钮。

直接看 dom 节点也能看出来：
![image](https://user-images.githubusercontent.com/13938334/191433975-059146b7-b005-4fef-981b-f6c697685096.png)

**Reply by Aaaaash** [reply]
https://github.com/opensumi/core/blob/main/packages/core-browser/src/react-providers/config-provider.tsx#L97-L102

ClientApp 的 constructor 里可以指定 `defaultPanels`，也即在没有其他缓存状态情况下的默认值
